akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -359,127 +359,117 @@ object ShardCoordinator {
  * @see [[ClusterSharding$ ClusterSharding extension]]
  */
 class ShardCoordinator(typeName: String, settings: ClusterShardingSettings,
-                       allocationStrategy: ShardCoordinator.ShardAllocationStrategy)
-  extends PersistentActor with ActorLogging {
+                       allocationStrategy: ShardCoordinator.ShardAllocationStrategy,
+                       replicator: ActorRef)
+  extends Actor with ActorLogging with Stash {
```

Comment:
In 2.4 we need to support both the old persistent implementation and the new ddata implementation. The reason is that akka-cluster-sharding is not experimental and therefore it cannot by default use the experimental distributed-data module.

Would it be possible to extract the specific persistent/ddata parts to separate subclasses? AbstractShardCoordinator, PersistentShardCoordinator, DDataShardCoordinator.